creed/tiling_utils_test.py

```diff
@@ -0,0 +1,745 @@
+    # 5. do not copy over non str, bool, int, or float values (ex. dict)
+    # 6. if a value in keys_ignore doesn't exist in input_dict, ignore
+    new_output_dict = tiling_utils.assign_metadata_vals(
+        example_input_dict, example_output_dict, example_keys_ignore
+    )
+
+    # assert the keys are correct
+    misc_utils.verify_same_elements(
+        new_output_keys=list(new_output_dict.keys()),
+        valid_keys=[1, 3, 4, 5, 6, 7]
+    )
+
+    # assert the values in each key is correct
+    assert new_output_dict[3] == 5.1
+    assert new_output_dict[4] == 1
+    assert 'hello' in new_output_dict[5] and new_output_dict[5]['hello'] == 'world'
+    assert new_output_dict[6] == 3.14
+    assert new_output_dict[7] is None
+
+
+def test_read_tiling_param(monkeypatch):
+    # test 1: int inputs
+    # test an invalid non-int response, an invalid int response, then a valid response
+    user_inputs_int = iter(['N', 0, 1])
+
+    # make sure the function receives the incorrect input first then the correct input
+    monkeypatch.setattr('builtins.input', lambda _: next(user_inputs_int))
+
+    # simulate the input sequence for int
+    sample_tiling_param = tiling_utils.read_tiling_param(
+        "Sample prompt: ",
+        "Sample error message",
+        lambda x: x == 1,
+        dtype=int
+    )
+
+    # assert sample_tiling_param was set to 1
+    assert sample_tiling_param == 1
+
+    # test 2: str inputs
+    # test an invalid non-str response, then an invalid str response, then a valid response
+    user_inputs_str = iter([1, 'N', 'Y'])
+
+    # make sure the function receives the incorrect input first then the correct input
+    monkeypatch.setattr('builtins.input', lambda _: next(user_inputs_str))
+
+    # simulate the input sequence for str
+    sample_tiling_param = tiling_utils.read_tiling_param(
+        "Sample prompt: ",
+        "Sample error message",
+        lambda x: x == 'Y',
+        dtype=str
+    )
+
+    # assert sample_tiling_param was set to 'Y'
+    assert sample_tiling_param == 'Y'
+
+
+def test_read_tiled_region_inputs(monkeypatch):
+    # define a sample fovs list
+    sample_fovs_list = test_utils.generate_sample_fovs_list(
+        fov_coords=[(0, 150), (100, 300)], fov_names=["TheFirstFOV", "TheSecondFOV"]
+    )
+
+    # define sample region_params to read data into
+    sample_region_params = {rpf: [] for rpf in settings.REGION_PARAM_FIELDS}
+
+    # intentionally place some lowercase letters as this function should support those
+    user_inputs = iter([2, 4, 1, 2, 'N', 4, 8, 2, 4, 'y'])
+
+    # override the default functionality of the input function
+    monkeypatch.setattr('builtins.input', lambda _: next(user_inputs))
+
+    # use the dummy user data to read values into the params lists
+    tiling_utils.read_tiled_region_inputs(
+        sample_fovs_list, sample_region_params
+    )
+
+    assert (sample_region_params[i]['region_start_x'] == 100 * i
```

Comment:
This should be a for loop. In addition, the values that are being evaluated should be passed from the function that creates the data, rather than being hardcoded here